src/components/StakingVaults/hooks/useFetchOpporunities.tsx

```diff
@@ -0,0 +1,94 @@
+import {
+  avalancheChainId,
+  bchChainId,
+  bscChainId,
+  btcChainId,
+  cosmosChainId,
+  dogeChainId,
+  ethChainId,
+  fromAccountId,
+  ltcChainId,
+} from '@shapeshiftoss/caip'
+import { useEffect, useState } from 'react'
+import { useSelector } from 'react-redux'
+import { useFeatureFlag } from 'hooks/useFeatureFlag/useFeatureFlag'
+import { setTimeoutAsync } from 'lib/utils'
+import { nftApi } from 'state/apis/nft/nftApi'
+import { zapper } from 'state/apis/zapper/zapperApi'
+import {
+  fetchAllOpportunitiesIdsByChainId,
+  fetchAllOpportunitiesMetadataByChainId,
+  fetchAllOpportunitiesUserDataByAccountId,
+} from 'state/slices/opportunitiesSlice/thunks'
+import {
+  selectPortfolioAccounts,
+  selectPortfolioAssetIds,
+  selectPortfolioLoadingStatus,
+  selectWalletAccountIds,
+} from 'state/slices/selectors'
+import { useAppDispatch } from 'state/store'
+
+export const useFetchOpportunities = () => {
+  const dispatch = useAppDispatch()
+  const portfolioLoadingStatus = useSelector(selectPortfolioLoadingStatus)
+  const requestedAccountIds = useSelector(selectWalletAccountIds)
+  const portfolioAssetIds = useSelector(selectPortfolioAssetIds)
+  const portfolioAccounts = useSelector(selectPortfolioAccounts)
+  const DynamicLpAssets = useFeatureFlag('DynamicLpAssets')
+
+  const [isLoading, setIsLoading] = useState(true)
+
+  // TODO: this needs to be split up into separate react-queries so we aren't refetching when moving around the app
+  useEffect(() => {
+    ;(async () => {
+      if (!requestedAccountIds.length) return
+      if (portfolioLoadingStatus === 'loading') return
+
+      setIsLoading(true)
+
+      dispatch(nftApi.endpoints.getNftUserTokens.initiate({ accountIds: requestedAccountIds }))
```

Comment:
Technically unrelated to opportunities - this made sense when in `<AppContext />` in one of our "master useEffects" but doesn't now 👀

Reply:
Although I suppose that's what the comment above implies?

Reply:
yeah agreed this could be separated, and it also hydrates portfolio/assets which would be related to global search. which wouldn't be taken into account as it stands.

Reply:
as discussed, lazy fetch of nfts implemented here 
5da9d911bb5554b7f239dfeafc583452added1f7